Portal/vip-application/src/main/java/fr/insalyon/creatis/vip/application/server/business/WorkflowBusiness.java

```diff
@@ -113,6 +115,31 @@ public WorkflowBusiness() {
         }
     }
 
+    private String selectEngineEndpoint(String applicationClass) throws BusinessException {
+        long min = 100000;
+        long tmp;
+        Engine engineBean = null;
+        String endpoint = "";
+        try {
+            List<Engine> availableEngines = ApplicationDAOFactory.getDAOFactory().getEngineDAO().getByClass(applicationClass);
+            for (Engine engine : availableEngines) {                
+                tmp = workflowDAO.getNumberOfRunningPerEngine(engine.getEndpoint());
+                if (tmp < min) {
+                    min = tmp;
+                    engineBean = engine;
+                }
+            }
+            endpoint = engineBean == null || engineBean.getEndpoint().isEmpty()
+                    ? Server.getInstance().getMoteurServer()
```

Comment:
I'm not sure if we are talking about the same think, but no, for me this is not a fallback.